components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/bean/SSOAgentConfig.java

```diff
@@ -192,8 +216,89 @@ public KeyStore getKeyStore() throws org.wso2.carbon.identity.sso.agent.exceptio
     public void setKeyStore(KeyStore keyStore) {
         this.keyStore = keyStore;
     }
+
     public void initConfig(Properties properties) throws SSOAgentException {
 
+        String decodedPassword;
+        boolean isReadpassword = false;
+        char[] password = null;
+
+        // Get copy of properties for looping in order to avoid ConcurrentModificationException.
+        Properties copyOfProperties = new Properties();
+        copyOfProperties.putAll(properties);
+
+        // Path of the password file.
+        String filePath = System.getProperty("catalina.home") + SSOAgentConstants.SSOAgentConfig.PASSWORD_FILEPATH;
+
+        // Looping through properties to check the encrypted property value by checking the prefix Enc:.
+        for (Map.Entry<Object, Object> entry : copyOfProperties.entrySet()) {
+            if (String.valueOf(entry.getValue()).startsWith("Enc:")) {
+                if (!System.getProperty(ARGUMENT).contains("password")) {
+
+                    // Check whether the password has been already read.
+                    if (!isReadpassword) {
+                        Path path = Paths.get(filePath);
+                        try (BufferedReader reader = Files.newBufferedReader(path, Charset.forName("UTF-8"))) {
+                            StringBuilder currentLine = new StringBuilder();
```

Comment:
currentLine is used in order to check whether file is empty or not. This variable has been used in order to avoid NPE when calling tocharArray.